src/Float/Extra.elm

```diff
@@ -408,3 +428,35 @@ in `Float.Extra.modBy modulus x`.
 modBy : Float -> Float -> Float
 modBy modulus x =
     x - modulus * toFloat (floor (x / modulus))
+
+
+{-| Interpolate from the first value to the second, based on a parameter that
+ranges from zero to one. Passing a parameter value of zero will return the start
+value and passing a parameter value of one will return the end value.
+
+    Float.Extra.interpolateFrom 5 10 0 == 5
+
+    Float.Extra.interpolateFrom 5 10 1 == 10
+
+    Float.Extra.interpolateFrom 5 10 0.6 == 8
+
+The end value can be less than the start value:
+
+    Float.Extra.interpolateFrom 10 5 0.1 == 9.5
+
+Parameter values less than zero or greater than one can be used to extrapolate:
+
+    Float.Extra.interpolateFrom 5 10 1.5 == 12.5
+
+    Float.Extra.interpolateFrom 5 10 -0.5 == 2.5
+
+    Float.Extra.interpolateFrom 10 5 -0.2 == 11
+
+-}
+interpolateFrom : Float -> Float -> Float -> Float
+interpolateFrom start end parameter =
+    if parameter <= 0.5 then
+        start + parameter * (end - start)
+
+    else
+        end + (1 - parameter) * (start - end)
```

Comment:
What's the reason for the if statement? I'd have thought that just `start + parameter * (end - start)` would be pretty standard `lerp`.

Reply:
My rationale was that `start + parameter * (end - start)` could (due to roundoff) lead to situations where `interpolateFrom a b 1.0` would _not_ give you exactly `b`, which seemed like it would be confusing/surprising. The version with the `if` _will_ recover `b` exactly in that case, since `1 - parameter` factor will be exactly zero. (It should also have better numerical accuracy for parameter values _close_ to 1.)

That said, I can't actually imagine a concrete use case where this would cause a problem.

Reply:
Of course that has to be weighed against the tiny performance cost of the extra comparison. I'd probably opt for removing it.